net.solarnetwork.node.hw.energymeter.mock/src/net/solarnetwork/node/datum/energymeter/mock/MockEnergyMeterDatumSource.java

```diff
@@ -0,0 +1,204 @@
+package net.solarnetwork.node.datum.energymeter.mock;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import org.springframework.context.MessageSource;
+
+import net.solarnetwork.node.DatumDataSource;
+import net.solarnetwork.node.domain.GeneralNodeACEnergyDatum;
+import net.solarnetwork.node.settings.SettingSpecifier;
+import net.solarnetwork.node.settings.SettingSpecifierProvider;
+import net.solarnetwork.node.settings.support.BasicTextFieldSettingSpecifier;
+import net.solarnetwork.node.settings.support.BasicToggleSettingSpecifier;
+
+/**
+ * Mock plugin to be the source of values for a GeneralNodeACEnergyDatum, the
+ * datum will contain random values based around a settable center deviating by
+ * a settable deviation.
+ * 
+ * <p>
+ * This class implements {@link SettingSpecifierProvider} and
+ * {@link DatumDataSource}
+ * </p>
+ * 
+ * @author robert
+ * @version 1.0
+ */
+
+public class MockEnergyMeterDatumSource implements DatumDataSource<GeneralNodeACEnergyDatum>, SettingSpecifierProvider {
+
+	@Override
+	public String getUID() {
```

Comment:
Would be great to make both `UID` and `groupUID` configurable & integrated as settings, so that other code can be configured to use this meter by those IDs. By convention we use a `uid` field with getter/setter methods, however. The [DatumDataSourceSupport](https://github.com/SolarNetwork/solarnetwork-node/blob/45338407ea7d81ea2a7afdcfa72eefcc10485314/net.solarnetwork.node/src/net/solarnetwork/node/support/DatumDataSourceSupport.java#L52) class can be extended to provide this functionality to this class, and you can use it's `getIdentifiableSettingSpecifiers()` method for settings support or just add similar ones to this class directly.